src/mailgun/mod.rs

```diff
@@ -17,28 +16,39 @@ pub struct MailgunClient {
     from: String,
 }
 
-impl From<crate::MailgunConfig> for MailgunClient {
-    fn from(config: crate::MailgunConfig) -> Self {
-        Self {
-            client: reqwest::Client::new(),
-            url: config.url,
-            api_user: config.user,
-            api_key: config.password,
-            default_to: config.to,
-            from: config.from,
+impl TryFrom<crate::MailgunConfig> for MailgunClient {
+    type Error = ();
+
+    fn try_from(config: crate::MailgunConfig) -> Result<Self, ()> {
+        if !config.disable {
+            || -> Option<_> {
+                Some(Self {
+                    url: config.url?,
+                    api_user: config.user?,
+                    api_key: config.password?,
+                    default_to: config.to?,
+                    from: config.from?,
+                    client: reqwest::Client::new(),
+                })
+            }()
+            .ok_or(())
+        } else {
+            Err(())
         }
```

Comment:
```suggestion
    fn try_from(config: crate::MailgunConfig) -> Result<Self, String> {
        if !config.disable {
            Ok(Self {
                url: config.url.ok_or("mailgun URL is not configured")?,
                api_user: config.user.ok_or("mailgun user is not configured")?,
                api_key: config.password.ok_or("mailgun password is not configured")?,
                default_to: config.to.ok_or("mailgun 'to' address is not configured")?,
                from: config.from.ok_or("mailgun 'from' address is not configured")?,
                client: reqwest::Client::new(),
            })
        } else {
            Err("mailgun is disabled".to_string())
        }
    }
```
This appears to fail the mailing configuration if the configuration is not set correctly. I think it's better to error here if the config is misconfigured 🤔